squire_lib/src/accounts.rs

```diff
@@ -63,19 +76,33 @@ pub struct OrganizationAccount {
 }
 
 impl SquireAccount {
-    /// Creates a new account object
-    pub fn new(user_name: String, display_name: String) -> Self {
-        Self {
-            user_name,
+    pub fn create(user_name: String, display_name: String, permissions: SharingPermissions) -> Self {
+        SquireAccount {
             display_name,
-            user_id: Uuid::new_v4().into(),
-            arena_name: None,
-            mtgo_name: None,
-            trice_name: None,
-            do_share: SharingPermissions::Everything,
+            user_name,
+            gamer_tags: HashMap::new(),
+            user_id: UserAccountID::new(Uuid::new_v4()),
```

Comment:
Again, this is stylist, but you can use `.into()` here. The type-ids are meant to prevent us from passing around the wrong object's id, but during object creation, we are creating a new Uuid, so there is no concern about that.
```suggestion
            user_id: Uuid::new_v4().into(),
```